dimod/decorators.py

```diff
@@ -197,28 +182,21 @@ def bqm_structured(f):
     function or method to accept a :obj:`.BinaryQuadraticModel` as the second
     input and for the :class:`.Sampler` to also be :class:`.Structured`.
     """
-
     @wraps(f)
-    def new_f(sampler, bqm, **kwargs):
-        try:
-            structure = sampler.structure
-            adjacency = structure.adjacency
-        except AttributeError:
-            if isinstance(sampler, Structured):
-                raise RuntimeError("something is wrong with the structured sampler")
-            else:
-                raise TypeError("sampler does not have a structure property")
-
-        if not all(v in adjacency for v in bqm.linear):
-            # todo: better error message
-            raise BinaryQuadraticModelStructureError("given bqm does not match the sampler's structure")
-        if not all(u in adjacency[v] for u, v in bqm.quadratic):
-            # todo: better error message
-            raise BinaryQuadraticModelStructureError("given bqm does not match the sampler's structure")
-
+    def structured_sample(sampler, bqm, **kwargs):
+        adjacency = sampler.adjacency
+
+        if not adjacency.keys() >= bqm.variables:
```

Comment:
Would `adjacency.keys() < bqm.variables` be better for readability?

Reply:
In this case they have a different meaning. The current syntax says "not superset", yours says "is subset".